probe-rs/src/architecture/riscv/communication_interface.rs

```diff
@@ -1791,6 +1795,11 @@ impl MemoryInterface for RiscvCommunicationInterface {
         self.write_multiple(address, data)
     }
 
+    fn write(&mut self, address: u64, data: &[u8]) -> Result<(), crate::Error> {
+        let address = valid_32_address(address)?;
```

Comment:
It checks that the address fits in 32bit. I renamed it to `valid_32bit_address` to be more concise. This is not necessarily much better, but I think with it's doc comment it's fine :)

I think this is because RISC-V does not support 64bit address space yet; at least in our codebase?